temporal/fx.go

```diff
@@ -278,19 +274,25 @@ func (s ServerFx) Stop() error {
 	return s.app.Stop(context.Background())
 }
 
-func StopService(logger log.Logger, app *fx.App, svcName primitives.ServiceName, stopChan chan struct{}) {
-	stopCtx, cancelFunc := context.WithTimeout(context.Background(), serviceStopTimeout)
+func StopService(
+	logger log.Logger,
+	app *fx.App,
+	svcName primitives.ServiceName,
+	stopChan chan struct{},
+	extraShutdownTimeFunc primitives.ExtraShutdownTimeFunc,
+) {
+	stopCtx, cancelFunc := context.WithTimeout(context.Background(), serviceStopTimeout+extraShutdownTimeFunc())
+	defer cancelFunc()
 	err := app.Stop(stopCtx)
-	cancelFunc()
 	if err != nil {
 		logger.Error("Failed to stop service", tag.Service(svcName), tag.Error(err))
 	}
 
 	// verify "Start" goroutine returned
 	select {
 	case <-stopChan:
-	case <-time.After(time.Minute):
-		logger.Error("Timed out (1 minute) waiting for service to stop.", tag.Service(svcName))
+	case <-stopCtx.Done():
+		logger.Error("Timed out waiting for service to stop", tag.Service(svcName))
```

Comment:
We don't know here how it can be changed, each service can decide that independently. We can list a few relevant dynamic configs in the error message but that might get out of sync with the code